frontend/src/components/Analysis/AnalysisReport/FineGrainedAnalysis/ComboAnalysisChart.tsx

```diff
@@ -70,12 +70,23 @@ export function ComboAnalysisChart(props: Props) {
     }
   }
 
-  let entryDataSum = 0;
-  entryData.forEach((entry) => (entryDataSum += entry[2]));
+  // Total samples for each true label
+  const entryDataSum: { [category: number]: number } = {};
+
+  /**
+   * `entryData` and `entryRatioData` are arrays of entries (arrays of length 3)
+   * each entry is [<true_label>, <predicted_label>, <value>]
+   * where <value> is sample count (integer) in `entryData`
+   * and ratio (float) in `entryRatioData`, respectively
+   */
+  entryData.forEach((entry) => {
+    const trueLabel = entry[0];
```

Comment:
You can use [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) to give proper names for each item in `entry`.

Something like `const [trueLabel, predictedLabel, value] = entry;` should work. The same goes for the `forEach` loop below. This also implicitly makes an assertion of the size of `entry`. You can also decide if the comment is still necessary after this renaming.